sklearn/preprocessing/data.py

```diff
@@ -1901,3 +1909,458 @@ def transform(self, X):
         """
         return _transform_selected(X, self._transform,
                                    self.categorical_features, copy=True)
+
+
+class QuantileNormalizer(BaseEstimator, TransformerMixin):
+    """Normalize features using quantiles information.
+
+    This Normalizer scales the features between 0 and 1, equalizing the
+    distribution of each feature to a uniform distribution. Therefore,
+    for a given feature, this normalization tends to spread out the most
+    frequent values. It also reduces the impact of (marginal) outliers:
+    this is therefore a robust preprocessing scheme.
+
+    The normalization is applied on each feature independently.
+    The cumulative density function of a feature is used to project the
+    original values. Features values of new/unseen data that fall below
+    or above the fitted range will be mapped to 0 and 1, respectively.
+    Note that this transform is non-linear. It may distort linear correlations
+    between variables measured at the same scale but renders variables measured
+    at different scales more directly comparable.
+
+    Parameters
+    ----------
+    n_quantiles : int, optional (default=1000)
+        Number of quantiles to be computed. It corresponds to the number
+        of landmarks used to discretize the cumulative density function.
+
+    subsample : int, optional (default=1e5)
+        Maximum number of samples used to estimate the quantiles.
+
+    ignore_implicit_zeros : bool, optional (default=False)
+        Apply only for sparse matrices. If True, the sparse entries of the
+        matrix are discarded to compute the quantile statistics. If false,
+        these entries are accounting for zeros.
+
+    output_pdf : str, optional (default='norm')
```

Comment:
Actually I think this parameter should be renamed "output_distribution" instead.